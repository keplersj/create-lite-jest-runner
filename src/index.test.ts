import { jest } from "@jest/globals";

describe("Create Lite Jest Runner", () => {
  it("exports as expected", async () => {
    const module_ = await import("./index");
    expect(module_).toMatchSnapshot();
  });

  describe("runner creation function", () => {
    it("creates a valid runner class when executed", async () => {
      const { default: create } = await import("./index");

      const runner = create(jest.fn());

      expect(runner).toMatchSnapshot();

      const instance = new runner({} as any);

      expect(instance).toMatchSnapshot();
      expect(instance.runTests).toMatchSnapshot();
    });
  });

  describe("runner class", () => {
    it("calls the onStart function and later onSuccess function when the runner function resolves", async () => {
      const { default: create } = await import("./index");
      const runnerFunction = jest.fn(() => Promise.resolve());
      const runnerClass = create(runnerFunction as any);

      const runnerInstance = new runnerClass({ maxWorkers: 1 } as any);

      const onStart = jest.fn();
      const onSuccess = jest.fn();
      const onFailure = jest.fn();

      await runnerInstance.runTests(
        [
          {
            context: {} as any,
            path: "/test.js",
          },
        ],
        undefined as any,
        onStart as any,
        onSuccess as any,
        onFailure as any
      );

      expect(runnerFunction.mock).toMatchSnapshot();
      expect(runnerFunction).toBeCalledTimes(1);
      expect(onStart.mock).toMatchSnapshot();
      expect(onStart).toBeCalledTimes(1);
      expect(onSuccess.mock).toMatchSnapshot();
      expect(onSuccess).toBeCalledTimes(1);
      expect(onFailure.mock).toMatchSnapshot();
      expect(onFailure).toBeCalledTimes(0);
    });

    it("calls the onStart function and later onFailure function when the runner function rejects", async () => {
      const { default: create } = await import("./index");
      const runnerFunction = jest.fn(() => Promise.reject("Test Throw"));
      const runnerClass = create(runnerFunction as any);

      const runnerInstance = new runnerClass({ maxWorkers: 1 } as any);

      const onStart = jest.fn();
      const onSuccess = jest.fn();
      const onFailure = jest.fn();

      await runnerInstance.runTests(
        [
          {
            context: {} as any,
            path: "/test.js",
          },
        ],
        undefined as any,
        onStart as any,
        onSuccess as any,
        onFailure as any
      );

      expect(runnerFunction.mock).toMatchSnapshot();
      expect(runnerFunction).toBeCalledTimes(1);
      expect(onStart.mock).toMatchSnapshot();
      expect(onStart).toBeCalledTimes(1);
      expect(onSuccess.mock).toMatchSnapshot();
      expect(onSuccess).toBeCalledTimes(0);
      expect(onFailure.mock).toMatchSnapshot();
      expect(onFailure).toBeCalledTimes(1);
    });
  });
});
