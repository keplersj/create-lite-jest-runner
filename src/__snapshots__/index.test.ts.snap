// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Lite Jest Runner exports as expected 1`] = `
Object {
  "default": [Function],
  "fail": [Function],
  "pass": [Function],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onFailure function when the runner function rejects 1`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "context": Object {},
        "testPath": "/test.js",
      },
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    5,
  ],
  "lastCall": Array [
    Object {
      "context": Object {},
      "testPath": "/test.js",
    },
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onFailure function when the runner function rejects 2`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "context": Object {},
        "path": "/test.js",
      },
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    4,
  ],
  "lastCall": Array [
    Object {
      "context": Object {},
      "path": "/test.js",
    },
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onFailure function when the runner function rejects 3`] = `
Object {
  "calls": Array [],
  "instances": Array [],
  "invocationCallOrder": Array [],
  "results": Array [],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onFailure function when the runner function rejects 4`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "context": Object {},
        "path": "/test.js",
      },
      "Test Throw",
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    6,
  ],
  "lastCall": Array [
    Object {
      "context": Object {},
      "path": "/test.js",
    },
    "Test Throw",
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onSuccess function when the runner function resolves 1`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "context": Object {},
        "testPath": "/test.js",
      },
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    2,
  ],
  "lastCall": Array [
    Object {
      "context": Object {},
      "testPath": "/test.js",
    },
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onSuccess function when the runner function resolves 2`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "context": Object {},
        "path": "/test.js",
      },
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    1,
  ],
  "lastCall": Array [
    Object {
      "context": Object {},
      "path": "/test.js",
    },
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onSuccess function when the runner function resolves 3`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "context": Object {},
        "path": "/test.js",
      },
      undefined,
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    3,
  ],
  "lastCall": Array [
    Object {
      "context": Object {},
      "path": "/test.js",
    },
    undefined,
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Create Lite Jest Runner runner class calls the onStart function and later onSuccess function when the runner function resolves 4`] = `
Object {
  "calls": Array [],
  "instances": Array [],
  "invocationCallOrder": Array [],
  "results": Array [],
}
`;

exports[`Create Lite Jest Runner runner creation function creates a valid runner class when executed 1`] = `[Function]`;

exports[`Create Lite Jest Runner runner creation function creates a valid runner class when executed 2`] = `
LiteTestRunner {
  "_context": Object {},
  "_globalConfig": Object {},
}
`;

exports[`Create Lite Jest Runner runner creation function creates a valid runner class when executed 3`] = `[Function]`;
